File: backend/src/main/scala/microservices/flight/objects/FlightDailyLowestPricesPlannerRequest.scala
// FlightDailyLowestPricesPlannerRequest defines the daily-lowest-prices planner request model.
package com.typesafe.travel.flight.objects

import io.circe.{Decoder, Encoder}
import io.circe.generic.semiauto.{deriveDecoder, deriveEncoder}

final case class FlightDailyLowestPricesPlannerRequest(
    departureAirport: String,
    arrivalAirport: String,
    startDate: String,
    days: Int,
    cabinClass: Option[String]
)

object FlightDailyLowestPricesPlannerRequest:
  given sourceEncoder: Encoder[FlightDailyLowestPricesPlannerRequest] = deriveEncoder
  given sourceDecoder: Decoder[FlightDailyLowestPricesPlannerRequest] = deriveDecoder

File: frontend/src/microservices/flight/objects/FlightDailyLowestPricesPlannerRequest.ts
// 本文件定义 flight 模块的 `FlightDailyLowestPricesPlannerRequest`，作为 planner 请求参数并提供 JSON 编解码。

export type FlightDailyLowestPricesPlannerRequest = {
  departureAirport: string
  arrivalAirport: string
  startDate: string
  days: number
  cabinClass?: string
}

export const flightDailyLowestPricesPlannerRequestFromJson = (json: string): FlightDailyLowestPricesPlannerRequest =>
  JSON.parse(json) as FlightDailyLowestPricesPlannerRequest

export const flightDailyLowestPricesPlannerRequestToJson = (value: FlightDailyLowestPricesPlannerRequest): string =>
  JSON.stringify(value)

